.github/workflows/sync_from_airtable.yml

```diff
@@ -73,38 +62,48 @@ jobs:
-                }
-                for (let i in assignees) {
-                    if (typeof ghLookup[assignees[i]] != 'undefined') {
-                    keepAssignees.push(assignees[i])
-                    delete ghLookup[assignees[i]];
+                const { removeAssignees, addAssignees } = assignees.reduce((acc, labelName, idx) => {
+                    const currentIdx = acc.priorAssignees.indexOf(labelName);
+
+                    if (currentIdx === -1) {
+                        acc.addAssignees.push(labelName);
                     } else {
-                    addAssignees.push(assignees[i]);
+                        // Keep the label indices from the priorLabel list
+                        acc.keepAssignees[currentIdx] = labelName;
                     }
+
+                    if (idx === labels.length - 1) {
+                        // We are in our final loop. Remove empty indices from keepAssignees and set those priorAssignees for removal.
+                        acc.keepAssignees = acc.keepAssignees.filter((label, keepLabelIdx) => {
+                            if (label === undefined) {
+                                acc.removeAssignees.push(acc.priorAssignees[keepLabelIdx]);
+                                return false;
+                            }
+                            
+                            return true;
+                        })
+                    }
+
+                    return acc;
+                }, { keepAssignees: new Array(issue.data.assignees.length), removeAssignees: [], addAssignees: [], priorAssignees: issue.data.assignees.map((assignee) => assignee.login)});
+
+                if (addAssignees.length > 0) {
+                    await github.rest.issues.addAssignees({
+                        owner: OWNER,
+                        repo: REPO,
+                        issue_number: NUMBER,
+                        assignees: addAssignees
+                    });
+                    core.info('Assignees Added: ', addAssignees.join(','))
                 }
-                const removeAssignees = Object.keys(ghLookup);
-                if (addAssignees && addAssignees.length > 0) {
-                  await github.rest.issues.addAssignees({
-                      owner: OWNER,
-                      repo: REPO,
-                      issue_number: NUMBER,
-                      assignees: addAssignees
-                  });
-                  core.info('Assignees Added: ', addAssignees.join(','))
-                }
-                if (removeAssignees && removeAssignees.length > 0) {
-                  await github.rest.issues.removeAssignees({
-                      owner: OWNER,
-                      repo: REPO,
-                      issue_number: NUMBER,
-                      assignees: removeAssignees
-                  });
-                  core.info('Assignees Removed: ', removeAssignees.join(','))
+                if (removeAssignees.length > 0) {
```

Comment:
Nitpick: for readability, a newline would help visually separate adding vs removing assignee blocks.